src/StateSmith/Output/Sim/SimWebGenerator.cs

```diff
@@ -148,9 +161,50 @@ private void CollectDiagramNames(StateMachine sm)
                         diagramEventNames.Add(trigger);
                 }
             }
+
+            // Collect available events for each state
+            if (vertex is State state)
+            {
+                var availableEvents = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                
+                // Collect events defined directly in this state
+                foreach (var behavior in state.Behaviors)
+                {
+                    foreach (var trigger in behavior.Triggers)
+                    {
+                        if (TriggerHelper.IsEvent(trigger))
+                        {
+                            availableEvents.Add(trigger);
+                        }
+                    }
+                }
+
+                // Collect events inherited from parent states
+                var currentVertex = state.Parent;
```

Comment:
can we remove the above foreach loop if we use the suggestion below?

```suggestion
                var currentVertex = state;
```